### PR TITLE
Kitsune arcsets pt 1: Simplify arc queries to use a single range

### PR DESCRIPTION
**Pull request by maackle**

### CODE REVIEW

File: crates/holochain_sqlite/src/sql/cell/fetch_op_hashes.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
+SELECT
```
[the remark this answers is not in the document]
Comment (maackle) [reply crates/holochain_sqlite/src/sql/cell/fetch_op_hashes.sql:1]: Oops, I missed a set of parens, I fixed it in a commit and now I'm seeing that it's using the indexes

[the remark this answers is not in the document]
Comment (maackle) [reply crates/holochain_sqlite/src/sql/cell/fetch_op_hashes.sql:1]: Although it's only using the timestamp index, not the storage_loc index, so maybe that's no good

[the remark this answers is not in the document]
Comment (freesig) [reply crates/holochain_sqlite/src/sql/cell/fetch_op_hashes.sql:1]: It really depends. Is the timestamp likely to shrink the search space a lot? If not you're better off using the storage index which won't work due to the or.
I don't really see the point to this tbh. 
This line `:storage_start_loc <= :storage_end_loc` can just be tested on the rust side and run a different sql query depending on the outcome. What is the reason for needing it in the sql?

**Review by freesig — APPROVED** [review]
NIcce

File: crates/kitsune_p2p/types/src/dht_arc.rs [hunk 2]
```diff
@@ -247,6 +247,24 @@ impl DhtArc {
...
+        let ArcRange { start, end } = self.range();
+        match (start, end) {
+            (Bound::Included(a), Bound::Included(b)) => Some((a, b)),
+            (Bound::Excluded(_), Bound::Excluded(_)) => None,
+            _ => unreachable!(),
+        }
+    }
+
+    /// Same as primitive_range, but "inside-out"
+    pub fn primitive_range_transposed(&self) -> (Option<u32>, Option<u32>) {
```
Comment (neonphog) [inline crates/kitsune_p2p/types/src/dht_arc.rs:262]: Just a semantic thing, but I don't like the name "transposed" it makes me think you're inverting the arc / selecting all the non-selected arc-space by swapping the start/end values.

Though, I'm having trouble coming up with a better name... maybe the two values are "detached" from each other?

```suggestion
    pub fn primitive_range_detached(&self) -> (Option<u32>, Option<u32>) {
```

File: crates/holochain_sqlite/src/sql/p2p_agent_store/gossip_query.sql [hunk 3]
```diff
@@ -2,33 +2,28 @@ SELECT agent
...
     (
-      :storage_start_1 IS NOT NULL
-      AND :storage_end_1 IS NOT NULL
-      AND storage_center_loc >= :storage_start_1
-      AND storage_center_loc <= :storage_end_1
+      -- wrapping case: everything *outside* the given range
+      :storage_start_loc > :storage_end_loc
+      AND (
+          storage_center_loc < :storage_end_loc
+          OR storage_center_loc > :storage_start_loc
```
Comment (neonphog) [inline crates/holochain_sqlite/src/sql/p2p_agent_store/gossip_query.sql:25]: Even though it wraps around, the marked points are still inclusive, so this needs to be <= / >= no?

```suggestion
          storage_center_loc <= :storage_end_loc
          OR storage_center_loc >= :storage_start_loc
```

**Review by neonphog — CHANGES_REQUESTED** [review]
inclusive wrapping range

**Review by neonphog — APPROVED** [review]
woot

**Pull request merged**